apps/system/lockscreen/js/component/lockscreen_basic_component.js

```diff
@@ -0,0 +1,162 @@
+/* global LockScreenStateLogger */
+'use strict';
+
+/**
+ * Component provides:
+ *
+ * 1. Resource keeper: to let all states share the same resources (cache).
+ * 2. Reference to the current activate state: so that parent component can
+ *    command and wait the sub-components to do things without tracking the
+ *    actual active state.
+ *
+ * Every states of this component would receive the Component instance as
+ * a way to access these common resources & properties. And every state
+ * transferring would done by the 'transferTo' method in this component,
+ * so that the component can update the active state correctly.
+ */
+(function(exports) {
+
+  /**
+   * View is the only thing parent component needs to manage.
+   */
+  var LockScreenBasicComponent = function(view) {
+    this._subcomponents = null;
+    this._activeState = null;
+    // Concrete components should extend these to let States access them.
+    // The first state component kick off should take responsibility for
+    // initializing these things.
+    //
+    // Resources is for external resources like settings value or DOM elements.
+    this.resources = {
+      elements: {
+        view: view
+      }
+    };
+    this.configs = {
+      logger: {
+        debug: false    // turn on it when we're debugging this component
+      }
+    };
+    this.logger = new LockScreenStateLogger();
+  };
+
+  /**
+   * State' phase is the component's phase.
+   */
+  LockScreenBasicComponent.prototype.phase =
```

Comment:
Why? It's just a simple getter needn't the dual setter, and people no need to get other phases except the 'current' one, so it's self-descriptive.